distribution/Dockerfile

```diff
@@ -0,0 +1,35 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+FROM centos:7
+
+# Project version defined in pom.xml is passed as an argument
+ARG VERSION
+
+# JDK 8 is a pre-requisite to run Drill ; 'which' package is needed for drill-config.sh
+RUN yum install -y java-1.8.0-openjdk-devel which ; yum clean all ; rm -rf /var/cache/yum
+
+# The drill tarball is generated upon building the Drill project
+COPY target/apache-drill-$VERSION.tar.gz /tmp
+
+# Drill binaries are extracted into the '/opt/drill' directory
+RUN mkdir /opt/drill
+RUN tar -xvzf /tmp/apache-drill-$VERSION.tar.gz --directory=/opt/drill --strip-components 1
+
+# Starts Drill in embedded mode and connects to Sqlline
+ENTRYPOINT /opt/drill/bin/drill-embedded
```

Comment:
Should we provide a `README.md` file? Explain that this particular `Dockerfile`:

* Runs an embedded Drillbit
* Works only for data already on the class path
* That the user must ssh(?) into the container?

Explain how to do additional configuration (as discussed in other comments).

Explain how to create a production `Dockerfile`.

Else, the user must be a bit of a Docker and Drill expert to work out what would be required, an it won't be clear what goal this particular file is trying to achieve.

Reply:
I do have a readme file - https://github.com/Agirish/drill/blob/docker_doc/docs/dev/Docker.md

I'll be adding that to this PR very shortly (once I clean-up).

Reply:
I also have some basic info in the description

Reply:
The PR description won't follow the code. Look forward to the readme file. I can point to it from the Docker section in the Drill book. (And, you'll have done the writing for me... Actually, by pointing to the readme, the reader can get the latest info, since this is a young feature and will change.)

Reply:
I've added the readme. I'm also working with @bbevens on a user facing doc. I'm not sure if there is anything else you would like me to add. And yes, surely we can update the doc as we add / modify features.